packages/time-series-chart/src/Chart.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from "react";
+import { LineChart, ChartProps } from "./LineChart";
+import { options, series } from "./SampleData";
+import { ComponentController, DataSet, ColumnType } from "@dashbuilder-js/component-api";
+import { useState, useEffect, useCallback } from "react";
+import { Options, SingleSeries } from "./Data";
+
+// Default Values
+const DEFAULT_WIDTH = 400;
+const DEFAULT_HEIGHT = 200;
+const NOT_ENOUGH_COLUMNS_MSG =
+  "Time series component expects 2 columns: category(LABEL or TEXT or NUMBER or DATE) and series(NUMBER).";
+const FIRST_COLUMN_INVALID_MSG = "Wrong type for first column, it should be either LABEL, TEXT, NUMBER or DATE.";
+const SECOND_COLUMN_INVALID_MSG = "Wrong type for second column, it should be NUMBER.";
+
+enum Params {
+  WIDTH = "width",
+  HEIGHT = "height"
+}
+
+enum AppStateType {
+  ERROR = "Error",
+  INIT = "Initializing",
+  LOADING_COMPONENT = "Loading Component",
+  LOADED_COMPONENT = "Loaded Component",
+  FINISHED = "Finished loading"
+}
+
+interface AppState {
+  state: AppStateType;
+  processesoptions: Options;
+  processesseries: Array<SingleSeries>;
+  configurationIssue: string;
+  message?: string;
+}
+
+const validateParams = (params: Map<string, any>): any => {
+  if (!params.get(Params.WIDTH)) {
+    return "Width is required.";
+  }
+  if (!params.get(Params.HEIGHT)) {
+    return "Height is required.";
+  }
+};
+
+const validateDataSet = (ds: DataSet): string | undefined => {
+  if (ds.columns.length < 2) {
+    return NOT_ENOUGH_COLUMNS_MSG;
+  }
+  if (
+    ds.columns[0].type !== ColumnType.LABEL &&
+    ds.columns[0].type !== ColumnType.TEXT &&
+    ds.columns[0].type !== ColumnType.NUMBER &&
+    ds.columns[0].type !== ColumnType.DATE
+  ) {
+    return FIRST_COLUMN_INVALID_MSG;
+  }
+  if (ds.columns[1].type !== ColumnType.NUMBER) {
+    return SECOND_COLUMN_INVALID_MSG;
+  }
+};
+
+interface Props {
+  controller: ComponentController;
+}
+
+function getseries(dataset: any): any {
```

Comment:
Please use the actual type on return, not any or we lose strong typing brought by typescript:

```
function getseries(dataset: DataSet): SingleSeries[] {
....
```